### ISSUE TITLE
Support PERCENT and WITH TIES in H2

### ISSUE THREAD

**Issue by katzyn** [jOOQ/jOOQ#8415]
Input:
```SQL
SELECT A FROM TEST ORDER BY A FETCH FIRST 10 PERCENT ROWS WITH TIES
```

Output:
```SQL
select A
from TEST
limit 10
```

Output is expected to be the same as input.

**Comment by lukaseder** [comment]
Thank you very much for your report. There are two problems here, one is a known problem: We currently do not emulate `PERCENT`: #7215

The other problem is that nothing gets emulated at all here for H2 (it works e.g. for PostgreSQL). Will fix that second part right away.

**Comment by katzyn** [comment]
Perhaps the translate tool should issue a warning when emulation is not possible.

However, this issue is not about emulation. Emulation is not needed here due to native support of both `PERCENT` and `WITH TIES` in the output dialect.

**Comment by lukaseder** [comment]
> Perhaps the translate tool should issue a warning when emulation is not possible.

Yes, that is going to be covered in #4448

> However, this issue is not about emulation. Emulation is not needed here due to native support of both PERCENT and WITH TIES in the output dialect.

Oh cool, sorry for the confusion. I had missed that!

**Comment by lukaseder** [comment]
For backwards compatibility reasons, we'll continue rendering `LIMIT .. OFFSET` in the absence of `PERCENT` and `WITH TIES` clauses.

**Comment by lukaseder** [comment]
Fixed for 3.12. Thanks again for your report!

**Comment by knutwannheden** [comment]
The changes don't cover the SQL Server style `TOP <n> [ PERCENT ] [ WITH TIES ]` which actually also is supported by H2. I will add a separate feature request for this.
